### DISCUSSION TITLE
Configuration of routing factors to improve drive time estimates based on country / region

### DISCUSSION THREAD

**Discussion by quicklybeforeitstoo** [organicmaps/organicmaps#6490] [Ideas]
In my experience in the US, OM consistently underestimates drive times in cities and suburbs, even under conditions of essentially no traffic. In particular, the drive time is typically at least 50% longer than predicted and sometimes close to 100% longer than predicted. (Of course it can be even more when there is significant traffic, but that is not what I am talking about.) Assuming it is more accurate in Europe, this may be due to different densities and timings of traffic lights, more use of stop signs and less use of roundabouts, etc. (This problem does not exist on highways.) While you could imagine trying to take into account each traffic light and stop sign, this would require a lot of data on these features to be downloaded. So, I am wondering whether the estimate could be improved significantly in a lighter-footprint way just by customizing the various routing factors per region. (I just started poking around the code so I don't have a concrete proposal of what would be necessary right now, but I see that various kinds of ways have different factors already.) Figuring out these factors for each region would require some actual timing data, but the current settings can be used as global defaults until user-provided data are validated. It would even be nice to have this as a configuration that users can tweak in-app.

As a side possibility, I wonder if this might provide a way to roughly account for traffic without live data. A different set of "rush hour" factors could be applied, perhaps manually, during busy times. (This is essentially what I already do in my head, but not all kinds of streets are affected equally and my intuition is worse outside of my home city.)

I know that these issues have been discussed before, (#1845 #3072  #3750 ) sometimes mixed with discussions about traffic, but I haven't seen much discussion of this in terms of regional differences, or in terms of parameters that could be exposed to the user. Curious what people think about this.

**Comment by biodranik** [comment]
As was already mentioned in other discussions, implementing traffic lights/turns/bumps penalties should greatly improve ETA. Any other improvements should be re-evaluated after this obvious missing "elephant in the room".

**Reply by patepelo** [reply]
ETA by car will always be very inaccurate without traffic information IMHO 

But penalties are still very important to improve routing, such as simplying the proposed route. 

@biodranik Should we start a separate discussion to post bad routing examples and try to analyze penaltiy times per different cases?

**Reply by biodranik** [reply]
Let's focus on the proper implementation of penalties first. Please update https://github.com/organicmaps/organicmaps/issues/1845 with relevant algorithm changes/suggestions and related test-cases where it can be tested. Note that the proposed route may change after incorporating penalties properly.
